src/models/StyledComponent.js

```diff
@@ -220,13 +226,30 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
         baseProps.ref = innerRef
       }
 
+      if (
+        process.env.NODE_ENV !== 'production' &&
+        this.props.sc &&
+        Object.prototype.toString.call(this.props.sc) !== '[object Object]' &&
```

Comment:
Yea, but an object is an easy false positive anyway. And that comes on top of this prop name being highly unlikely in my opinion as everyone would assume it being related to us.

So if we assume an object we just constrain the user and we’re not reliably detecting misuse anyway :/ seems like a suboptimal compromis

Reply:
>  prop name being highly unlikely in my opinion as everyone would assume it being related to us

Maybe. I've learned never to assume things in this industry, haha.

> Yea, but an object is an easy false positive anyway.

Not sure I follow. There's no surefire way to detect if it's being used correctly, but at least verifying it's an object seems like an OK plan over doing nothing. I originally was thinking about trying to detect if the prop was used inside an interpolation, but that would be hard to do without wrapping it in an ES6 Proxy, which would be a compatibility nightmare.

Reply:
Well, you could define a getter on the interpolation props we are generating, at least in dev. But I disagree that doing sth here is better than nothing ;) this would unnecessarily restrict the API to lock out theming/type switches that are already really popular :)

Reply:
Do you have an opinion on this @mxstbr?